ert_shared/ensemble_evaluator/evaluator.py

```diff
@@ -223,22 +231,46 @@ async def count_dispatcher(self):
     async def handle_dispatch(self, websocket, path):
         async with self.count_dispatcher():
             async for msg in websocket:
-                event = from_json(
-                    msg, data_unmarshaller=serialization.evaluator_unmarshaller
-                )
+                try:
+                    event = from_json(
+                        msg, data_unmarshaller=serialization.evaluator_unmarshaller
+                    )
+                except cloudevents.exceptions.DataUnmarshallerError:
+                    event = from_json(msg, data_unmarshaller=lambda x: pickle.loads(x))
                 await self._dispatch.handle_event(self, event)
                 if event["type"] in [
                     identifiers.EVTYPE_ENSEMBLE_STOPPED,
                     identifiers.EVTYPE_ENSEMBLE_FAILED,
                 ]:
                     return
 
+    async def handle_result(self, websocket, path):
+        if self._result is None:
+            event = CloudEvent(
+                {
+                    "type": identifiers.EVTYPE_EE_RESULT_NOT_READY,
+                    "source": f"/ert/ee/{self._ee_id}",
+                },
+            )
+        else:
+            event = CloudEvent(
+                {
+                    "type": identifiers.EVTYPE_EE_RESULT,
+                    "source": f"/ert/ee/{self._ee_id}",
+                    "datacontenttype": "application/octet-stream",
+                },
+                cloudpickle.dumps(self._result),
+            )
+        await websocket.send(to_json(event))
+
     async def connection_handler(self, websocket, path):
         elements = path.split("/")
         if elements[1] == "client":
             await self.handle_client(websocket, path)
         elif elements[1] == "dispatch":
             await self.handle_dispatch(websocket, path)
+        elif elements[1] == "result":
+            await self.handle_result(websocket, path)
```

Comment:
Should this be followed by an `else` raising an exception for unhandled case?

Reply:
Don't think we should raise at least since that will probably give some 5XX error code back. What happens now is that the websocket connection is just not accepted, since it is not a valid path. Think this behaviour is fine, but we could probably log on the server side that somebody made a request to an unknown path.

Reply:
Logging sounds good 👍🏻